cpp/Platform.Equality/Platform.Equality.Any.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#ifndef PLATFORM_EQUALITY_ANY
+#define PLATFORM_EQUALITY_ANY
+
+#include <any>
+#include <concepts>
+#include <stdexcept>
+#include <typeindex>
+
+#include "Platform.Equality.EqualityComparerFunctionType.h"
+
+namespace Platform::Equality
+{
+    namespace Internal
+    {
+        template<class TValue, class TEqualityComparer>
+        inline auto ToAnyEqualVisitor(TEqualityComparer&& func)
+        {
+            return std::pair<std::type_index, EqualityComparerFunctionType<std::any>>{
+                std::type_index(typeid(TValue)),
+                [func = std::forward<TEqualityComparer>(func)](const std::any& left, const std::any& right) -> std::size_t
+                {
+                    if constexpr (std::is_void_v<TValue>)
```

Comment:
what is this check for?

Reply:
It allows to execute func without arguments?

Reply:
something like?
```cpp
RegisterEqualityComparer<void>([]() { /*...*/ });
```
Wouldn't it be better to just disallow the use of `void` on concepts? In cpp we cannot create an object of `void` type

Reply:
How do we disallow it?

Reply:
```cpp
template<class TValue, class TEqualityComparer>
  requires std::is_object_v<TValue>
inline auto ToAnyEqualVisitor(TEqualityComparer&& func)
```
^ Template will not be instantiated for non-objects